src/modules/AdvancedPaste/AdvancedPaste/Helpers/ClipboardHelper.cs

```diff
@@ -21,9 +20,28 @@ internal static void SetClipboardTextContent(string text)
 
             if (!string.IsNullOrEmpty(text))
             {
+                Logger.LogDebug("Text to set to clipboard: " + text);
+
                 DataPackage output = new();
                 output.SetText(text);
-                Clipboard.SetContentWithOptions(output, null);
+                bool success = Clipboard.SetContentWithOptions(output, null);
+                Logger.LogInfo("Setting Clipboard data was success? : " + success);
+
+                // Wait 50 ms in a loop until setting the clipboard content has really finished.
+                while (!Clipboard.GetContent().GetTextAsync().GetAwaiter().GetResult().Contains(text))
+                {
+                    Thread.Sleep(50);
+                }
+
+                try
+                {
+                    string clipContent = Clipboard.GetContent().GetTextAsync().GetAwaiter().GetResult();
+                    Logger.LogInfo("Clipboard content for current process:" + clipContent);
```

Comment:
let us use debug log. Thoughts?

And I missed a space after the ":" to separate text and clipboard content.

Reply:
I think, yes, we should go ahead and use debug log here.

Reply:
Do you think we should replace it below as well?
```cs
DataPackage output = new();
output.SetText(text);
bool success = Clipboard.SetContentWithOptions(output, null);
Logger.LogInfo("Setting Clipboard data was success? : " + success);
```

Reply:
I have replaced both of them in the latest commit. Let me know if we need any more changes.

Reply:
As we have no explicit error log it's better to use info log for the "set clipboard" result. We don't log debug level on the machines of our users.

I suggested the debug log for the clipboard content to set for two reasons:
1. Privacy. We should not log every clipboard data especially without errors.
2. Keep log file smaller.

Reply:
I have replaced them with Debug since that information is only needed when we're actually debugging the code. Also, we don't want to sneak peek at the clipboard content whenever the user calls the module.